salt/_extensions/pillar/ca.py

```diff
@@ -295,46 +296,203 @@ def get_ca_signed_cert(cacert_path, ca_name, CN):
+                    print(f"Key file not found: {key_file}")
+                    continue
+
+                with cert_file.open('r') as f_cert:
+                    cert_data = f_cert.read()
+
+                with key_file.open('r') as f_key:
+                    key_data = f_key.read()
+
+                # Store combined certificate and key
+                combined_data = "\n".join([cert_data, key_data])
+                acme_certs[domain_name] = combined_data
+                # print(f"read certificate for {domain_name}")
+
+            except Exception as e:
+                print(f"Error processing certificate for {domain_dir.name}: {e}")
+    
+    except Exception as e:
+        print(f"Error scanning ACME certificates directory: {e}")
+
+    print(f"Found {len(acme_certs)} ACME certificates")
+    return acme_certs
+
+
+def _process_ca_certificates(minion_id, pillar, base="/etc/ssl", name="PSFCA", cert_opts=None):
+    ca_data = {
+        "ca": {},
+        "certs": {},
+    }
+
+    try:
+        if cert_opts is None:
+            cert_opts = {}
+
+        # Create CA certificate
+        opts = cert_opts.copy()
+        opts["CN"] = name
+        create_ca(base, name, **opts)
+
+        ca_data["ca"][name] = get_ca_cert(base, name)
+
+        # Process CA-signed certificates (gen_certs)
+        gen_certs = pillar.get("tls", {}).get("gen_certs", {})
+        for certificate, config in gen_certs.items():
+            role_patterns = [
+                role.get("pattern")
+                for role in [
+                    pillar.get("roles", {}).get(r) for r in config.get("roles", "")
+                ]
+                if role and role.get("pattern") is not None
+            ]
+
+            if any(compound(pat, minion_id) for pat in role_patterns):
+                # Create the options
+                opts = cert_opts.copy()
+                opts["CN"] = certificate
+                opts["days"] = config.get("days", 1)
+
+                create_ca_signed_cert(base, name, **opts)
+
+                # Add the signed certificates to the pillar data
+                cert_data = get_ca_signed_cert(base, name, certificate)
+                ca_data["certs"][certificate] = cert_data
+    except Exception as e:
+        print(f"Error processing CA certificates: {e}")
+
+    return ca_data
 
-    # Ensure we have a CA created.
-    opts = cert_opts.copy()
-    opts["CN"] = name
-    create_ca(base, name, **opts)
 
-    # Start our pillar with just the ca certificate.
+def _process_acme_certificates(minion_id, pillar):
+    """Process ACME certificates
+    
+    Reads ACME certificates and determines which ones should be available
+    to the specified minion based on access rules.
+    """
+    acme_certs = {}
+    
+    try:
+        print(f"Processing ACME certificates for minion: {minion_id}")
+        all_acme_certs = _find_acme_certs()
+        
+        # Check if this is a loadbalancer (gets all certs)
```

Comment:
This is all a lot of work, when we could theoretically just assign certs explicitly to roles, see other comment.